runtime/frame_descriptors.c

```diff
@@ -93,7 +93,12 @@ static caml_frame_descrs build_frame_descriptors(link* frametables)
         h = (h+1) & table.mask;
       }
       table.descriptors[h] = d;
-      d = next_frame_descr(d);
+      /* Don't traverse to next frame unnecessarily. If last frame has
+         frame_size=-1 then we will be reading link-time determined bytes
```

Comment:
The function `next_frame_descr` already contains a test for `frame_size = -1` in order to decide whether to skip over debug info or not. To avoid handling this case in two different places, perhaps `next_frame_descr` could just check whether `frame_size = -1` at the top and return `NULL` in that case.

Reply:
Agree that the fix should be in `next_frame_descr`. 

> check whether frame_size = -1 at the top and return NULL in that case

seems the right thing to do.